info.go

```diff
@@ -0,0 +1,113 @@
+package chi
+
+import (
+	"go/parser"
+	"go/token"
+	"os"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"strings"
+)
+
+type FuncInfo struct {
```

Comment:
I'd prefer to keep these function-introspective functions in the `docgen` package.